test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -150,9 +152,168 @@ public void HtmlLocalizer_HtmlWithInvalidResourcestring_ThrowsException(string f
+            var actualLocalizedString = htmlLocalizer.GetString("John");
+
+            // Assert
+            Assert.Equal("Hello John", actualLocalizedString.Value);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_GetStringWithArguments_ReturnsLocalizedString()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer.GetString("John", "Doe");
+
+            // Assert
+            Assert.Equal("Hello John Doe", actualLocalizedString.Value);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_Html_ReturnsLocalizedHtmlString()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer.Html("John");
+
+            // Assert
+            Assert.Equal("Hello John", actualLocalizedString.Value);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_WithCulture_ReturnsLocalizedHtmlString()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer.WithCulture(new CultureInfo("fr"))["John"];
+
+            // Assert
+            Assert.Equal("Bonjour John", actualLocalizedString.Value);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_GetAllStrings_ReturnsAllLocalizedStrings()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var allLocalizedStrings = htmlLocalizer.GetAllStrings(false).ToList();
+
+            //Assert
+            Assert.Equal(1, allLocalizedStrings.Count);
+            Assert.Equal("World", allLocalizedStrings.First().Value);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_GetAllStringsIncludeAncestorCulture_ReturnsAllLocalizedStrings()
+        {
+            // Arrange
+            var stringLocalizer = new TestStringLocalizer();
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer, new CommonTestEncoder());
+
+            // Act
+            var allLocalizedStrings = htmlLocalizer.GetAllStrings().ToList();
+
+            //Assert
+            Assert.Equal(2, allLocalizedStrings.Count);
+            Assert.Equal("World", allLocalizedStrings[0].Value);
+            Assert.Equal("Bar", allLocalizedStrings[1].Value);
+        }
     }
 
     public class TestClass
     {
     }
+
+    public class TestStringLocalizer : IStringLocalizer
```

Comment:
These are not nested classes. I moved them to separate files. I am using `TestStringLocalizer` in these tests and also in `ViewLocalizerTest`